### PR TITLE
add more info to adhd example

### LINKED ISSUE #3737: [DOC] improve instructions in "Default Mode Network extraction of ADHD dataset" example
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed suggestion in detail.

It seems the instructions in [this example](https://nilearn.github.io/dev/auto_examples/04_glm_first_level/plot_adhd_dmn.html#default-mode-network-extraction-of-adhd-dataset) need some improvement. There was a confusion mentioned on [NeuroStar](https://neurostars.org/t/why-is-there-glm-for-resting-state-data/25841). After discussing with @Remi-Gau, we concluded that maybe we can add one or two lines saying that in this example we extract the activity of a seed region and then use the extracted signal as regressor in a GLM and this will yield the correlation of each region with the seed region.

### List any pages that would be impacted.

https://nilearn.github.io/dev/auto_examples/04_glm_first_level/plot_adhd_dmn.html#default-mode-network-extraction-of-adhd-dataset

### PR DESCRIPTION
**Pull request by mtorabi59**
Closes #3737 .

Changes proposed in this pull request:

- add a plot of the signal extracted from the seed region
- add more explanation about what the example will show

### CODE REVIEW

**Comment by bthirion** [comment]
LGTM, thx ! Would you like to take a look at the black failure ?

**Comment by Remi-Gau** [comment]
re running the failed job: don't see why it should fail in this PR. I suspect something flaky.

**Comment by Remi-Gau** [comment]
I am also getting it on #3742 so I suspect this a new thing and unrelated to this PR.

Will try to see where this is coming from before we merge those new PRs.

**Comment by mtorabi59** [comment]
> I am also getting it on #3742 so I suspect this a new thing and unrelated to this PR.
> 
> Will try to see where this is coming from before we merge those new PRs.

Thanks @Remi-Gau !!

File: examples/04_glm_first_level/plot_adhd_dmn.py [hunk 1]
```diff
@@ -56,6 +59,22 @@
 )
 seed_time_series = seed_masker.fit_transform(adhd_dataset.func[0])
 frametimes = np.linspace(0, (n_scans - 1) * t_r, n_scans)
+
+#########################################################################
+# Plot the time course of the seed region.
+import matplotlib.pyplot as plt
+
+fig = plt.figure(figsize=(15, 5))
```
Comment (bthirion) [inline examples/04_glm_first_level/plot_adhd_dmn.py:67]: Can you make the figure ---much--- smaller ? For the moment, the labels, title and legend are dwarfed by the figsize.

**Review by bthirion — APPROVED** [review]
LGTM, Thx !

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
@ymzayek I think this one can go in too. WDYT?
The pre-release dep is unrelated.
There was a failure on 3.9 (see below) but reran the workflow and it passed.

https://github.com/nilearn/nilearn/actions/runs/5159236549/jobs/9293907646#step:8:14481
```
FAILED nilearn/plotting/tests/test_matrix_plotting.py::test_sanitize_reorder_error[None] - AssertionError: Regex pattern did not match.
 Regex: "Parameter reorder needs to be one of {False, True, 'complete', 'average', 'single'}"
 Input: "Parameter reorder needs to be one of {False, True, 'single', 'average', 'complete'}."
FAILED nilearn/plotting/tests/test_matrix_plotting.py::test_sanitize_reorder_error[foo] - AssertionError: Regex pattern did not match.
 Regex: "Parameter reorder needs to be one of {False, True, 'complete', 'average', 'single'}"
 Input: "Parameter reorder needs to be one of {False, True, 'single', 'average', 'complete'}."
FAILED nilearn/plotting/tests/test_matrix_plotting.py::test_sanitize_reorder_error[2] - AssertionError: Regex pattern did not match.
 Regex: "Parameter reorder needs to be one of {False, True, 'complete', 'average', 'single'}"
 Input: "Parameter reorder needs to be one of {False, True, 'single', 'average', 'complete'}."
= 3 failed, 2581 passed, 7 skipped, 2 xfailed, 1300 warnings in 1644.62s (0:27:24) =
```

**Comment by ymzayek** [comment]
Ok @Remi-Gau I'll look over it today.

> There was a failure on 3.9 (see below) but reran the workflow and it passed.

Yes I've seen this failure a few times this week. I haven't looked into it yet but we can open an issue to discuss it

File: examples/04_glm_first_level/plot_adhd_dmn.py [hunk 2]
```diff
@@ -39,9 +42,9 @@
 pcc_coords = (0, -53, 26)
 
 #########################################################################
-# Estimate contrasts
+# Extract the seed region's time course
 # ------------------
```
Comment (ymzayek) [inline examples/04_glm_first_level/plot_adhd_dmn.py:46]:
```suggestion
# Extract the seed region's time course
# -------------------------------------
```

to resolve the sphinx warning: title underline is too short

**Review by ymzayek — COMMENTED** [review]
@mtorabi59 looks great, just a minor thing to make sure the sphinx full build will build correctly on main.

**Pull request merged**